mpa/cls/trainer.py

```diff
@@ -65,8 +65,12 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
                     dash_line)
 
         # Data
-        if 'unlabeled' in cfg.data:
+        if 'unlabeled' in cfg.data and cfg.train_type == 'SEMISUPERVISED':
             datasets = [[build_dataset(cfg.data.train), build_dataset(cfg.data.unlabeled)]]
+            datasets[0][0].samples_per_gpu = cfg.data.samples_per_gpu
+            datasets[0][0].workers_per_gpu = cfg.data.workers_per_gpu
+            datasets[0][1].samples_per_gpu = cfg.data.unlabeled.get('samples_per_gpu', cfg.data.samples_per_gpu*3)
```

Comment:
This is likely to cause index errors. As far as I know, there may be no unlabeled dataset.

Reply:
+ It looks good to be able to receive the unlabeled batch size from otx. (not a fixed 3x value)
It seems inconvenient to have to edit this code to fix this (unlabeled batch size).

Reply:
Yes, please share your ideas if you have good suggestions. Same mechanism can be applied on other tasks, I think. @supersoob, @jaegukhyun

Reply:
Agree with @harimkang and @jaegukhyun. I think we should find a way to adapt batch size for unlabeled dataset. And I also used unlabeled data hook for seg. I hope you go over this part. I don't think it doesn't seem good to load composedDL by checking the dataset length.

Reply:
These code lines are removed, please check updated code to support unlabeled batch size. Thank you.